conf.py

```diff
@@ -165,8 +167,21 @@
 # sphinx-notfound-page
 notfound_urls_prefix = "/"
 
+# Dynamically expose the Python version associated with the "main" branch.
+# Exactly one entry in ``release-cycle.json`` should have ``"branch": "main"``.
+with open("include/release-cycle.json", encoding="UTF-8") as _f:
+    _cycle = json.load(_f)
+
+_main_entries = [v for v, d in _cycle.items() if d.get("branch") == "main"]
+if len(_main_entries) != 1:
+    raise RuntimeError(
+        "release-cycle.json must contain exactly one entry with 'branch': 'main'"
+    )
+
+_main_version = _main_entries[0]
```

Comment:
We'll always have a `main` so can simplify:
```suggestion
main_version = next(version for version, data in _cycle.items() if data.get("branch") == "main")
```